### PR TITLE
[maca] adjust rotary_embedding.

### PR DESCRIPTION
**Pull request by Reinerzhou**

### CODE REVIEW

File: dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu [hunk 1]
```diff
@@ -128,31 +161,34 @@ void rotary_embedding(
     torch::Tensor& key,    // [batch_size, seq_len, num_kv_heads * head_size] or
                            // [num_tokens, num_kv_heads * head_size]
     int64_t head_size,
-    torch::Tensor& cos_sin_cache,  // [max_position, rot_dim]
+    torch::Tensor& cos,  // [max_position, embed_dim]
+    torch::Tensor& sin,  // [max_position, embed_dim]
     bool is_neox) {
+  auto cos_sin_cache = torch::cat({cos, sin}, -1);
```
Comment (jinminxi104) [inline dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu:167]: cos_sin_cache is unused

File: dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu [hunk 2]
```diff
@@ -128,31 +161,34 @@ void rotary_embedding(
...
+
   int64_t num_tokens = query.numel() / query.size(-1);
-  int rot_dim = cos_sin_cache.size(1);
+  int embed_dim = cos.size(-1);
```
Comment (jinminxi104) [inline dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu:170]: what is the shape of cos/sin and cos_sin_cache?

File: dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu [hunk 3]
```diff
@@ -114,7 +147,7 @@ __global__ void batched_rotary_embedding_kernel(
   const scalar_t* cache_ptr =
       cos_sin_cache + (cos_sin_cache_offset + pos) * rot_dim;
 
-  apply_rotary_embedding<scalar_t, IS_NEOX>(
+  batched_rotary_embedding<scalar_t, IS_NEOX>(
```
Comment (jinminxi104) [inline dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu:150]: 名字改错了？

File: dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu (hunk 2 above)
Comment (Reinerzhou) [reply dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu:170]: cos/sin: [max_position, embed_dim ]
cos_sin_cache:  [max_position, rot_dim]
embed_dim  = rot_dim / 2
only embed_dim  is used in kernel.

File: dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu (hunk 3 above)
Comment (Reinerzhou) [reply dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu:150]: before:

- void rotary_embedding  -> rotary_embedding_kernel  ->inline __device__ void apply_rotary_embedding.
- void batched_rotary_embedding-> batched_rotary_embedding_kernel-> inline __device__ void apply_rotary_embedding.



now:

- Due to changes in the  inference of inline __device__ void apply_rotary_embedding.
- void rotary_embedding  -> rotary_embedding_kernel  ->inline __device__ void apply_rotary_embedding.
- void batched_rotary_embedding-> batched_rotary_embedding_kernel-> inline __device__ void batched_rotary_embedding(just renamed).

File: dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu (hunk 1 above)
Comment (Reinerzhou) [reply dlinfer/vendor/maca/csrc/pos_encoding_kernels.cu]: removed.

**Pull request merged**